indexd/index/drivers/alchemy.py

```diff
@@ -171,21 +191,26 @@ def hashes_to_urls(self, size, hashes, start=0, limit=100):
 
             return [r.url for r in query]
 
-    def add(self, form, size=None, urls=[], hashes={}, did=None):
+
+    def add(self, form, size = None, urls = [], hashes = {}, did = None, baseid = None):
         '''
         Creates a new record given urls and hashes.
         '''
 
         with self.session as session:
             record = IndexRecord()
+            base_version = None
 
-            if did is None:
-                did = str(uuid.uuid4())
-            record.did = did
-            record.rev = str(uuid.uuid4())[:8]
+            if(baseid == None):
```

Comment:
Generally this would be written as `if baseid is None`—space after `if`, no parens, and using `is None` for comparison
(same thing with `if` style elsewhere)